homeassistant/components/synology_srm/config_flow.py

```diff
@@ -46,21 +35,18 @@ async def async_step_user(self, user_input=None):
             try:
                 client = get_srm_client_from_user_data(user_input)
                 device_id = await self.hass.async_add_executor_job(
-                    _login_and_fetch_device_id, client
+                    fetch_srm_device_id, client
```

Comment:
What is the `device_id` actually ?

A MAC adress, serial number, ...

Reply:
`device_id` can be compared with a serial number. For instance, it's in the following format `synology_irr436p_rt2600ac`. It's much easier to fetch it from the API with a simple endpoint and the device ID is also written in the back of the device so it's a good device ID to display to the user IMO.